### PR TITLE
I think this fixes the device_type, role and platform index issues in module 3 when running 3.6.6

### PR DESCRIPTION
**Pull request by mrmrcoleman**
I also updated `device_role` to the new name of `role` and a bunch of other small fixes in the instructions

### CODE REVIEW

**Comment by mrmrcoleman** [comment]
@richbibby I also added a bunch of small text edits in you should take a look through before merging

**Review by richbibby — APPROVED** [review]

**Pull request merged**
